utils/src/const_value.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    fmt::{Debug, Display},
+    hash::{Hash, Hasher},
+    ops::Deref,
+};
+
+/// A wrapper for a type that is not supposed to be modified, no matter one
+/// This particularly solves the problem of not being able to mark member variables as const in Rust
+pub struct ConstValue<T> {
```

Comment:
I understand this wrapper works on best effort basis. It blocks calling `&mut self` methods on the inner value.

As far as I can see, one can still do something like:
```rust
fn modify_const_value(cv: &mut ConstValue<u32>) {
    *cv = ConstValue::new(42);
}
```

Same applies if you're given a `&mut` of a struct containing a `ConstValue`. Also anything with interior mutability can still be modified.